pkg/apis/v1alpha1/cluster_supply_chain.go

```diff
@@ -254,27 +144,58 @@ type SupplyChainResource struct {
 	Configs []ResourceReference `json:"configs,omitempty"`
 }
 
-var ValidSupplyChainTemplates = []client.Object{
-	&ClusterSourceTemplate{},
-	&ClusterImageTemplate{},
-	&ClusterConfigTemplate{},
-	&ClusterTemplate{},
-}
-
 type SupplyChainTemplateReference struct {
 	// Kind of the template to apply
 	//+kubebuilder:validation:Enum=ClusterSourceTemplate;ClusterImageTemplate;ClusterTemplate;ClusterConfigTemplate
 	Kind string `json:"kind"`
+
+	// Name of the template to apply
+	// Only one of Name and Options can be specified.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+
+	// Options is a list of template names and Selectors. The templates must all be of type Kind.
+	// A template will be selected if the workload matches the specified Selector.
+	// Only one template can be selected.
+	// Only one of Name and Options can be specified.
+	// +kubebuilder:validation:MinItems=2
+	Options []TemplateOption `json:"options,omitempty"`
+}
+
+type TemplateOption struct {
 	// Name of the template to apply
 	// +kubebuilder:validation:MinLength=1
 	Name string `json:"name"`
+
+	// Selector is a field query over a workload resource.
+	Selector Selector `json:"selector"`
 }
 
-type SupplyChainStatus struct {
-	Conditions         []metav1.Condition `json:"conditions,omitempty"`
-	ObservedGeneration int64              `json:"observedGeneration,omitempty"`
+type Selector struct {
+	// MatchFields is a list of field selector requirements. The requirements are ANDed.
+	// +kubebuilder:validation:MinItems=1
+	MatchFields []FieldSelectorRequirement `json:"matchFields"`
+}
+
+type FieldSelectorRequirement struct {
+	// Key is the JSON path in the workload to match against.
```

Comment:
Given the definition of `validWorkloadPath`, it seems like we could make more explicit that the possible set of JSONPath constructs here is restricted to the spec 

```suggestion
	// Key is the JSON path in the Workload spec to match against.
```

Reply:
The term spec is overloaded. We handle more than just workload.spec. workload.metadata is also valid.

Reply:
Oh shoot, forgot to add workload.metadata to prefix only list

Reply:
I dont know how to suggest a suggestion on a suggested edit, so: how about "Key is the JSON path into the Workload Resource to match against"